File: packages/core/src/internal/InternalMetrics.js
export class InternalMetrics {
  constructor() {
    this.eventsDropped = 0;
    this.logsDropped = 0;
    this.cardinalityDropped = 0;
    this.framesSent = 0;
    this.framesFailed = 0;
    this.bytesUncompressed = 0;
    this.bytesCompressed = 0;
    this.eventsBuffered = 0;
    this.logsBuffered = 0;
    this.lastSyncMs = 0;
    this.configVersion = 0;
    this.processRssBytes = 0;
  }

  hasCounterActivity() {
    return (
      this.eventsDropped !== 0 ||
      this.logsDropped !== 0 ||
      this.cardinalityDropped !== 0 ||
      this.framesSent !== 0 ||
      this.framesFailed !== 0 ||
      this.bytesUncompressed !== 0 ||
      this.bytesCompressed !== 0
    );
  }

  snapshotAndReset() {
    const snap = {
      eventsDropped: this.eventsDropped,
      logsDropped: this.logsDropped,
      cardinalityDropped: this.cardinalityDropped,
      framesSent: this.framesSent,
      framesFailed: this.framesFailed,
      bytesUncompressed: this.bytesUncompressed,
      bytesCompressed: this.bytesCompressed,
      eventsBuffered: this.eventsBuffered,
      logsBuffered: this.logsBuffered,
      lastSyncMs: this.lastSyncMs,
      configVersion: this.configVersion,
      processRssBytes: this.processRssBytes
    };
    this.eventsDropped = 0;
    this.logsDropped = 0;
    this.cardinalityDropped = 0;
    this.framesSent = 0;
    this.framesFailed = 0;
    this.bytesUncompressed = 0;
    this.bytesCompressed = 0;
    return snap;
  }
}

File: clients/csharp/Runtime/Core/InternalMetrics.cs
namespace Wardx
{
    public sealed class InternalMetrics
    {
        public double EventsDropped;
        public double LogsDropped;
        public double CardinalityDropped;
        public double FramesSent;
        public double FramesFailed;
        public double BytesUncompressed;
        public double BytesCompressed;
        public double EventsBuffered;
        public double LogsBuffered;
        public double LastSyncMs;
        public double ConfigVersion;
        public double ProcessRssBytes;

        public bool HasCounterActivity()
        {
            return EventsDropped != 0
                || LogsDropped != 0
                || CardinalityDropped != 0
                || FramesSent != 0
                || FramesFailed != 0
                || BytesUncompressed != 0
                || BytesCompressed != 0;
        }

        public InternalSnapshot SnapshotAndReset()
        {
            var snap = new InternalSnapshot
            {
                EventsDropped = EventsDropped,
                LogsDropped = LogsDropped,
                CardinalityDropped = CardinalityDropped,
                FramesSent = FramesSent,
                FramesFailed = FramesFailed,
                BytesUncompressed = BytesUncompressed,
                BytesCompressed = BytesCompressed,
                EventsBuffered = EventsBuffered,
                LogsBuffered = LogsBuffered,
                LastSyncMs = LastSyncMs,
                ConfigVersion = ConfigVersion,
                ProcessRssBytes = ProcessRssBytes
            };
            EventsDropped = 0;
            LogsDropped = 0;
            CardinalityDropped = 0;
            FramesSent = 0;
            FramesFailed = 0;
            BytesUncompressed = 0;
            BytesCompressed = 0;
            return snap;
        }
    }

    public sealed class InternalSnapshot
    {
        public double EventsDropped;
        public double LogsDropped;
        public double CardinalityDropped;
        public double FramesSent;
        public double FramesFailed;
        public double BytesUncompressed;
        public double BytesCompressed;
        public double EventsBuffered;
        public double LogsBuffered;
        public double LastSyncMs;
        public double ConfigVersion;
        public double ProcessRssBytes;
    }
}
